File: programmers/같은숫자는싫어.py
from sys import stdin

def solution(arr):
  answer = [arr[0]]
  idx = arr[0]
  for num in arr[1:]:
    if idx == num:
      pass
    else:
      idx = num
      answer.append(num)
    
  return answer

def init():
  arr = list(map(int, stdin.readline().split()))
  print(solution(arr))
  
init()

File: js_algorithm/programmers/1Level/같은숫자는싫어.js
function solution(arr) {
  const answer = [];
  arr.reduce((prev, curr) => {
    if (prev !== curr) {
      answer.push(curr);
    }
    return curr;
  }, -1);

  return answer;
}

function init() {
  const array = [
    [1, 1, 3, 3, 0, 1, 1],
    [4, 4, 4, 3, 3],
  ];
  for (const arr of array) {
    console.log(solution(arr));
  }
}

init();
